platform-ajax/src/test/java/com/softicar/platform/ajax/testing/selenium/engine/common/AjaxSeleniumTestServerWaiterTest.java

```diff
@@ -0,0 +1,84 @@
+package com.softicar.platform.ajax.testing.selenium.engine.common;
+
+import com.softicar.platform.ajax.testing.selenium.engine.level.low.AbstractAjaxSeleniumLowLevelTest;
+import com.softicar.platform.dom.elements.DomDiv;
+import com.softicar.platform.dom.event.IDomBlurEventHandler;
+import com.softicar.platform.dom.event.IDomClickEventHandler;
+import com.softicar.platform.dom.event.IDomEvent;
+import com.softicar.platform.dom.event.IDomFocusEventHandler;
+import com.softicar.platform.dom.event.IDomInputEventHandler;
+import com.softicar.platform.dom.input.DomTextInput;
+import java.util.ArrayList;
+import java.util.List;
+import org.junit.Test;
+
+public class AjaxSeleniumTestServerWaiterTest extends AbstractAjaxSeleniumLowLevelTest {
+
+	private final TestDiv testDiv;
+	private final List<String> events;
+
+	public AjaxSeleniumTestServerWaiterTest() {
+
+		this.testDiv = openTestNode(TestDiv::new);
+		this.events = new ArrayList<>();
+	}
+
+	@Test
+	public void test() {
+
+		send(testDiv.input1, "a");
+		click(testDiv.input2);
+		waitForServer();
```

Comment:
Shouldn't we test `AjaxSeleniumTestServerWaiter` here directly? Like `new AjaxSeleniumTestServerWaiter(webDriver).waitForServer()`. Otherwise, we kind of assume an implementation detail of `AbstractAjaxSeleniumLowLevelTest.waitForServer`.

Reply:
changed